src/IonBinaryWriter.ts

```diff
@@ -654,18 +654,15 @@ class IntNode extends LeafNode {
   constructor(writer: LowLevelBinaryWriter, parent: Node, annotations: Uint8Array, private readonly value: number | JSBI) {
     super(writer, parent, IonTypes.INT, annotations);
 
-    if (this.value === 0) {
-      this.intTypeCode = TypeCodes.POSITIVE_INT;
-      this.bytes = new Uint8Array(0);
-    } else if (this.value > 0) {
+    if (this.value > 0) {
       this.intTypeCode = TypeCodes.POSITIVE_INT;
       let writer: LowLevelBinaryWriter = new LowLevelBinaryWriter(new Writeable(LowLevelBinaryWriter.getUnsignedIntSize(this.value)));
       writer.writeUnsignedInt(this.value);
       this.bytes = writer.getBytes();
-    } else {
+
+    } else if (this.value < 0) {
       this.intTypeCode = TypeCodes.NEGATIVE_INT;
       let magnitude: number | JSBI;
-      let uintSize: number;
       if (value instanceof JSBI) {
         if(JsbiSupport.isNegative(value)) {
           magnitude = JSBI.unaryMinus(value);
```

Comment:
I am pretty sure we should be using `JSBI.GT` and `JSBI.LT` instead of the operator.  In fact, I am a bit surprised that this works, do we have tests that invoke this code path with `JSBI` values?

Should the preamble of this method check for `typeof this.value === 'number' || this.value isinstance JSBI` as per Wes's feedback to avoid strange behavior for abstract  comparisons?

Reply:
I've added binary `writeInt` tests that pass JSBI values.  These new tests pass locally when the implementation uses `>`/`<` operators as well as `JSBI.GT`/`JSBI.LT` calls; not sure why.  I agree the latter approach feels safer.

I've also added a check to assert that `this.value` is a number or JSBI; we might want to have a discussion about whether similar assertions would make sense in other parts of the impl.